### PR TITLE
Android settings (AEC)

### PR DESCRIPTION
**Pull request by gcalero**
This PR adds android application a settings screen (Menu > Settings) **only in debug builds**.
The only setting by the moment is Acoustic Echo Cancellation: it's a feature present in some android phones like Google Pixel 2, Samsung S8, Samsung S9. Other devices like Huawei Mate 9 pro don't support it.

AEC must be disabled by default (we want to test its performance before turning it on by default). The user has to enable it through Menu > Settings > AEC
AEC must be disabled when a headset is plugged in (1)

Test plan
=======
- Check that AEC setting is disabled by default
- Check audio quality with regards to master
- Try starting HighFidelity app in both condition: plugged and unplugged headset
- Switch AEC setting and check that the right input device is used each time
- Plug/unplug the headset and check that that the right input device is used each time
- Check that echo cancelation is enabled (2)
- Check that AEC option only appears in debug builds (not in release ones)

(1) You can check input device switching in logcat, simply filter by "[hifi.audioclient] switchInputToAudioDevice". **voicerecognition** is an opensl es preset that uses the audio input with minimum or no processing (without AEC). **voicecommunication** is a preset that uses AEC and, if supported, noise reduction too.
(2) Echo cancelation performance depends on the device implementation. It may cancel echo at max volume or not. In all cases you can appreciate a difference between AEC enabled and disabled (voicecommunication and voicerecognition).

### CODE REVIEW

**Comment by hifi-gustavo** [comment]
Android build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dandroid/12926/HighFidelity-Beta-PR13981-2943502.apk

**Comment by hifi-gustavo** [comment]
Mac OS X build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dmacos/12926/HighFidelity-Beta-PR13981-2943502.dmg

**Comment by hifi-gustavo** [comment]
Mac OS X client-only build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dmacos/12926/HighFidelity-Beta-Interface-PR13981-2943502.dmg

**Comment by hifi-gustavo** [comment]
Windows build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dwindows/12926/HighFidelity-Beta-PR13981-2943502.exe

**Comment by hifi-gustavo** [comment]
Windows client-only build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dwindows/12926/HighFidelity-Beta-Interface-PR13981-2943502.exe

**Comment by hifi-gustavo** [comment]
Mac OS X build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dmacos/12955/HighFidelity-Beta-PR13981-27c7592.dmg

**Comment by hifi-gustavo** [comment]
Mac OS X client-only build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dmacos/12955/HighFidelity-Beta-Interface-PR13981-27c7592.dmg

**Comment by hifi-gustavo** [comment]
Windows build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dwindows/12955/HighFidelity-Beta-PR13981-27c7592.exe

**Comment by hifi-gustavo** [comment]
Windows client-only build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dwindows/12955/HighFidelity-Beta-Interface-PR13981-27c7592.exe

**Comment by gcalero** [comment]
build this please

**Comment by hifi-gustavo** [comment]
Android build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dandroid/12958/HighFidelity-Beta-PR13981-27c7592.apk

**Comment by hifi-gustavo** [comment]
Mac OS X build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dmacos/12958/HighFidelity-Beta-PR13981-27c7592.dmg

**Comment by hifi-gustavo** [comment]
Mac OS X client-only build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dmacos/12958/HighFidelity-Beta-Interface-PR13981-27c7592.dmg

**Comment by hifi-gustavo** [comment]
Windows build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dwindows/12958/HighFidelity-Beta-PR13981-27c7592.exe

**Comment by hifi-gustavo** [comment]
Windows client-only build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dwindows/12958/HighFidelity-Beta-Interface-PR13981-27c7592.exe

**Comment by hifi-gustavo** [comment]
Windows build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dwindows/12963/HighFidelity-Beta-PR13981-ec2c022.exe

**Comment by hifi-gustavo** [comment]
Windows client-only build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dwindows/12963/HighFidelity-Beta-Interface-PR13981-ec2c022.exe

File: android/app/src/main/java/io/highfidelity/hifiinterface/MainActivity.java [hunk 1]
```diff
@@ -108,6 +111,16 @@ protected void onCreate(Bundle savedInstanceState) {
         }
     }
 
+    private void updateDebugMenu(Menu menu) {
+        if (BuildConfig.DEBUG) {
```
[the remark this answers is not in the document]
Comment (gcalero) [reply android/app/src/main/java/io/highfidelity/hifiinterface/MainActivity.java]: Talked with Samuel. We will test how it works in debug build and then decide

File: libraries/audio-client/src/AudioClient.cpp [hunk 2]
```diff
@@ -451,9 +451,14 @@ QAudioDeviceInfo defaultAudioDeviceForMode(QAudio::Mode mode) {
 
 #if defined (Q_OS_ANDROID)
     if (mode == QAudio::AudioInput) {
+        Setting::Handle<bool> enableAEC(SETTING_AEC_KEY, false);
+        bool aecEnabled = enableAEC.get();
+        auto audioClient = DependencyManager::get<AudioClient>();
+        bool headsetOn = audioClient? audioClient->isHeadsetPluggedIn() : false ;
```
[the remark this answers is not in the document]
Comment (gcalero) [reply libraries/audio-client/src/AudioClient.cpp]: Fixed

**Comment by hifi-gustavo** [comment]
Android build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dandroid/12971/HighFidelity-Beta-PR13981-1215ddb.apk

**Comment by hifi-gustavo** [comment]
Windows build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dwindows/12971/HighFidelity-Beta-PR13981-1215ddb.exe

**Comment by hifi-gustavo** [comment]
Windows client-only build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dwindows/12971/HighFidelity-Beta-Interface-PR13981-1215ddb.exe

**Comment by hifi-gustavo** [comment]
Mac OS X build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dmacos/12971/HighFidelity-Beta-PR13981-1215ddb.dmg

**Comment by hifi-gustavo** [comment]
Mac OS X client-only build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dmacos/12971/HighFidelity-Beta-Interface-PR13981-1215ddb.dmg

**Comment by hifi-gustavo** [comment]
Android build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dandroid/12977/HighFidelity-Beta-PR13981-5795ef3.apk

**Comment by qliacoombs** [comment]
Build this please

**Comment by hifi-gustavo** [comment]
Android build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dandroid/13210/HighFidelity-Beta-PR13981-5795ef3.apk

**Comment by hifi-gustavo** [comment]
Mac OS X build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dmacos/13210/HighFidelity-Beta-PR13981-5795ef3.dmg

**Comment by hifi-gustavo** [comment]
Mac OS X client-only build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dmacos/13210/HighFidelity-Beta-Interface-PR13981-5795ef3.dmg

**Comment by hifi-gustavo** [comment]
Windows build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dwindows/13210/HighFidelity-Beta-PR13981-5795ef3.exe

**Comment by hifi-gustavo** [comment]
Windows client-only build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dwindows/13210/HighFidelity-Beta-Interface-PR13981-5795ef3.exe

**Comment by qlimnavarro** [comment]
Currently Testing

**Comment by qlimnavarro** [comment]
Currently Testing

**Comment by qlimnavarro** [comment]
Pass @conklin94122 
Settings found in logcat, AEC is default and audio sounds good.

**Pull request merged**